src/main/java/com/atlassian/bitbucket/jenkins/internal/config/BitbucketSearchEndpoint.java

```diff
@@ -47,9 +48,10 @@ public HttpResponse doFindProjects(
             @Nullable @QueryParameter("credentialsId") String credentialsId,
             @Nullable @QueryParameter("name") String name) {
         Jenkins.get().checkPermission(CONFIGURE);
+        BitbucketServerConfiguration serverConf = getServer(serverId);
         BitbucketProjectSearchClient projectSearchClient =
                 bitbucketClientFactoryProvider
-                        .getClient(getServer(serverId), getCredentials(credentialsId))
+                        .getClient(serverConf.getBaseUrl(), BitbucketCredentialsAdaptor.createWithFallback(credentialsId, serverConf))
```

Comment:
`serverConf.getBaseUrl()` is annotated as `@Nullable` but the client requires it to be nonnull